pkg/userdata/amzn2/provider.go

```diff
@@ -302,6 +311,14 @@ write_files:
   content: |
 {{ .ContainerRuntimeConfig | indent 4 }}
 
+{{- if and (eq .ContainerRuntimeName "docker") (ne .ContainerRuntimeAuthConfig "{}") }}
```

Comment:
`(ne .ContainerRuntimeAuthConfig "{}")` should return an empty string instead of `{}` if no credentials were provided